Base/QTGUI/qSlicerDataDialog.cxx

```diff
@@ -389,6 +411,28 @@ bool qSlicerDataDialogPrivate::propagateChange(int changedRow)const
     && (QApplication::keyboardModifiers() & Qt::ShiftModifier);
 }
 
+//-----------------------------------------------------------------------------
+bool qSlicerDataDialogPrivate::checkAndHandleArchive(const QFileInfo& file)
+{
+  if (file.suffix() == "zip")
+    {
+    if (QMessageBox::question(this, tr("Open archive?"), tr("The selected file is a .zip archive, open it and load contents?")))
+      {
+      this->temporaryArchiveDirectory.reset(new QTemporaryDir());
+      if (this->temporaryArchiveDirectory->isValid())
+        {
+        // C function in vtkArchive
+        if (unzip(file.absoluteFilePath().toStdString().c_str(), this->temporaryArchiveDirectory->path().toStdString().c_str()))
+          {
+          this->addDirectory(QDir(this->temporaryArchiveDirectory->path()));
```

Comment:
Yes, that's true.  Unfortunately all of the loading is done in the coreIOManager and by that time there's no way to tell it that these are from a temp directory.  The only workaround I can think of would be to iterate through the storage nodes after the load and change the ones that correspond to the unzipped data, but that seems ugly.

I don't think the current behavior is so bad, because by the time a user goes to save they will have to decide where they want to put the data anyway, and they have control.

Reply:
Let's let Murat test this out for his use cases and suggest what would be good behavior.